text/0034-colored-coins/0034-colored-coins.md

```diff
@@ -0,0 +1,359 @@
+                                Set to <strong>value 0</strong> to denote an <i>Ed25519 Address</i>.
+                            </td>
+                        </tr>
+                        <tr>
+                            <td>Address</td>
+                            <td>Array&lt;byte&gt;\[32\]</td>
+                            <td>The raw bytes of the Ed25519 address which is a BLAKE2b-256 hash of the Ed25519 public key.</td>
+                        </tr>
+                    </table>
+                </details>
+            </td>
+        </tr>
+	        <tr>
+            <td>Color</td>
+            <td>Array&lt;byte&gt;[33] </td>
+            <td>The color of tokens of this <i>SigLockedColoredOutput</i> output.
+			</td>
+        </tr>
+        <tr>
+            <td>Amount</td>
+            <td>uint64</td>
+            <td>The amount of tokens of this <i>SigLockedColoredOutput</i> output.</td>
+        </tr>
+    </table>
+
+<p></p>
+
+There are couple new concepts as compared to a `SigLockedSingleOutput`.
+
+## Color
+
+Colors are globally unique and are assigned to tokens upon minting. A color refers to an output in a transaction that
+minted it, therefore colors are essentially `OutputIDs`. Since an `OutputID` is the combination of the `TransactionID`
+and the `index of the output within the transaction`, it is possible to locate the genesis transaction of a color by
+taking the first 32 bytes of it.
+
+The zero value of a color (33 bytes that are all zero) has a distinct name, it is called `ColorIOTA`, and defines the
+color of an IOTA token.
+
+## Opcodes
+
+Opcodes, or operation codes, define how the protocol treats the output during transaction validation and post processing.
+In general, we can divide up the processing of a transaction into 3 phases:
+- **Syntactical validation**: validating rules that govern the transaction layout and can be objectively decided just
+  by looking at the raw transaction.
+- **Semantic validation**: validation rules that are concerned with the content of the inputs, outputs and unlock blocks.
+  Valid unlock blocks are a semantic check for example.
+- **Post Processing**: Once both of the previous validations are passed, the created outputs need to be booked into the
+  ledger. This step involves calculating the `OutputIDs`, and additionally can have other actions depending on the opcode.
+
+In the previous version of the IOTA Protocol, opcodes did not exist, because the only operation the protocol allowed was
+actually to `MOVE` tokens by unlocking them from an input and locking them into a new output.
+With colored coins however, the ability to specify `MINT` operation for the creation of colors, and `UNCOLOR` operation
+to destroy such colors is absolutely necessary.
+
+Opcodes affect the 3 transaction processing stages:
+
+### Syntactical Validation With Opcodes
+
+The `Opcode` field of a `SigLockedColoredOutput` may only be:
+
+- `0` to denote `MINT` operation,
```

Comment:
Would it maybe make sense to have `MOVE` as the "default operation"?
I would set it to 0, because if there is somewhere a bug in a client lib or 3rd party lib, not setting the opcode field, maybe already colored tokes would become "worthless" because they get another color.
And `MOVE` was the only possible operation in the old IOTA :shrug:  :)

Reply:
Good idea! @muXxer what do you think about the new [version](https://github.com/iotaledger/protocol-rfcs/pull/34/commits/24b300652244d2047e855b787e2984cc88fc922a)?